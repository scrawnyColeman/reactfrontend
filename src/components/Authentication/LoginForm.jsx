import React, { useState } from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '../Button/component';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import { materialStyles } from '../../constants/styles';
import { useHistory } from 'react-router-dom';
import AuthenticationService from '../AuthenticationService/AuthenticationService.js';

const LoginForm = () => {
    const useStyles = makeStyles(materialStyles);
    const history = useHistory();
    const classes = useStyles();
    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');

    return (
        <form className={classes.root} noValidate autoComplete="off">
            <Grid container spacing={3}>
                <Grid item xs={12}>
                    <TextField
                        id="outlined-username"
                        name="Username"
                        type="text"
                        variant="filled"
                        className={classes.textfield}
                        onChange={(e) => setUsername(e.target.value)}
                    />
                </Grid>
            </Grid>
            <Grid container spacing={3}>
                <Grid item xs={12}>
                    <TextField
                        id="outlined-password"
                        type="password"
                        autoComplete="current-password"
                        variant="filled"
                        className={classes.textfield}
                        onChange={(e) => setPassword(e.target.value)}
                    />
                </Grid>
            </Grid>
            <Grid container spacing={3}>
                <Grid item xs={12}>
                    <Button
                        text="Log in"
                        variant="text"
                        onClick={() => {
                            /**
                             * DO BACKEND
                             */
                            if (username === 'scrawnycoleman' && password === 'dummy') {
                                history.push({ pathname: `profile/${username}` });
                                AuthenticationService.registerLogin(username, 'sample@email.com', 1);
                            }
                        }}
                    />
                </Grid>
            </Grid>
        </form>
    );
};

export default LoginForm;
