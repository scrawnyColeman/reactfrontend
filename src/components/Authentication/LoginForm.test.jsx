import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import LoginForm from './LoginForm';

Enzyme.configure({ adapter: new Adapter() });

describe('LoginForm component', () => {
    test('component matches snapshot', () => {
        const wrapper = shallow(<LoginForm />);
        expect(wrapper).toMatchSnapshot();
    });
});
