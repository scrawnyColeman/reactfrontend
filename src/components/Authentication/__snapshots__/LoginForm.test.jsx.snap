// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component component matches snapshot 1`] = `
<form
  autoComplete="off"
  className="makeStyles-root-1"
  noValidate={true}
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(TextField))
        id="outlined-username"
        label="Username"
        type="text"
        variant="filled"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(TextField))
        autoComplete="current-password"
        id="outlined-password"
        label="Password"
        type="password"
        variant="filled"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <Button
        text="Log in"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</form>
`;
